src/select.rs

```diff
@@ -87,3 +92,40 @@ impl<'a> TElement<'a> for &'a ElementData {
         }
     }
 }
+
+impl<'a> Node<'a> {
+    pub fn css(&'a self, css_str: &str) -> FilterNodes<'a, Descendants<'a>> {
+        let selectors = match parser::parse_author_origin_selector_list_from_str(css_str) {
+            Ok(vec) => vec,
+            Err(_) => Vec::new(),
+        };
+        FilterNodes{
+            iter: self.descendants(),
+            phantom: PhantomData,
+            filter: selectors,
+        }
+    }
+}
+
+pub struct FilterNodes<'a, T:'a>
+    where T: Iterator<Item=&'a Node<'a>>{
+    iter: T,
+    phantom: PhantomData<&'a T>,
```

Comment:
Is this required? I think you could remove it if you remove the `'a` parameter and the `where` clause as well. (The `where` clause is only really required on the `Iterator` impl.) Compare with [`std::iter::Filter`](https://github.com/rust-lang/rust/blob/c21fd9a34f763c532ded7bf141b90a01cef76c43/src/libcore/iter.rs#L1741-L1744) for example.